### PR TITLE
Add centerVertical option to Modal

### PR DESCRIPTION
**Pull request by adhayward**
I was attempting to make my modals vertically centered as per the [bootstrap docs](https://getbootstrap.com/docs/4.4/components/modal/#vertically-centered) and discovered there was no way to add the appropriate class via this widget so I thought I'd add one.

This PR adds two new properties to the Modal widget:
1. centerVertical - a boolean that when true includes the modal-dialog-centered class. Defaults to false for BC

2. dialogOptions - an array for the options on the modal-dialog div. The classes that were previously added to this element are now added to dialogOptions via Html::addCssClass

| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | Y
| Fixed issues  |

### CODE REVIEW

**Comment by simialbi** [comment]
I miss a test here

**Comment by adhayward** [comment]
I've added tests now and also included an option for modal-dialog-scrollable too as that can easily work in the same way as modal-dialog-centered.

**Comment by simialbi** [comment]
Nice. Ty. If you introduce `$dialogOptions` anyway, could you add the `role` parameter the same way it's done in `$options`?

File: src/Modal.php [hunk 1]
```diff
@@ -125,9 +143,18 @@ public function init()
 
         $this->initOptions();
 
+        Html::addCssClass($this->dialogOptions, 'modal-dialog');
```
Comment (simialbi) [inline src/Modal.php:146]: I'd move this to `initOptions` method.

File: src/Modal.php [hunk 2]
```diff
@@ -125,9 +143,18 @@ public function init()
...
+        Html::addCssClass($this->dialogOptions,  $this->size);
+        if ($this->centerVertical) {
+            Html::addCssClass($this->dialogOptions,  'modal-dialog-centered');
+        }
+        if ($this->scrollable) {
+            Html::addCssClass($this->dialogOptions,  'modal-dialog-scrollable');
+        }
```
Comment (simialbi) [inline src/Modal.php:153]: I'd move this to `initOptions` method.

**Comment by adhayward** [comment]
@simialbi I've moved those lines to initOptions now and added the role="document" in.

The role addition does mean that the default output is different to before - does it count as breaking BC? As it's a correction to bring it inline with the bootstrap docs and shouldn't have any major effects I would assume it's ok but probably worth mentioning.

**Comment by samdark** [comment]
Adding role is not a big deal in terms of breaking anything. It's fine to have it without major version bump. A mention in release notes is a good idea.

File: src/Modal.php [hunk 3]
```diff
@@ -115,7 +115,25 @@ class Modal extends Widget
      */
     public $toggleButton = false;
 
+    /**
+     * @var boolean whether to center the modal vertically
+     *
+     * When true the modal-dialog-centered class will be added to the modal-dialog
```
Comment (samdark) [inline src/Modal.php:121]: Need `@since 2.0.9`

File: src/Modal.php [hunk 4]
```diff
@@ -115,7 +115,25 @@ class Modal extends Widget
...
+     */
+    public $centerVertical = false;
+    /**
+     * @var boolean whether to make the modal body scrollable
+     *
+     * When true the modal-dialog-scrollable class will be added to the modal-dialog
```
Comment (samdark) [inline src/Modal.php:127]: Need `@since 2.0.9`

File: src/Modal.php [hunk 5]
```diff
@@ -115,7 +115,25 @@ class Modal extends Widget
...
+     */
+    public $scrollable = false;
 
+    /**
+     * @var array modal dialog options
```
Comment (samdark) [inline src/Modal.php:132]: Need `@since 2.0.9`

**Comment by adhayward** [comment]
@samdark those should now be included

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**